skrub/_interpolation_joiner.py

```diff
@@ -177,9 +177,9 @@ def __init__(
         self,
         aux_table,
         *,
+        key=None,
```

Comment:
This is less of a trivial change here. I let @jeromedockes judging this one.

Reply:
the order proposed in this pr is the same as polars and pandas so  I think it is better